Q1.

Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

<u>Answer</u>:

To calculate the Pearson correlation coefficient between the amount of time students spend studying for an exam and their final exam scores, you would need to first compute the covariance between the two variables and their standard deviations.

Let's assume you have the following data:

| Student | Study time (hours) | Exam score (out of 100) |
|---------|--------------------|-------------------------|
| 1       | 5                  | 85                      |
| 2       | 2                  | 60                      |
| 3       | 6                  | 90                      |
| 4       | 4                  | 75                      |
| 5       | 3                  | 70                      |

To calculate the Pearson correlation coefficient, you would first need to compute the means and standard deviations of each variable:

mean(study time) = (5 + 2 + 6 + 4 + 3) / 5 = 4
mean(exam score) = (85 + 60 + 90 + 75 + 70) / 5 = 76

sd(study time) = sqrt(((5-4)^2 + (2-4)^2 + (6-4)^2 + (4-4)^2 + (3-4)^2) / 4) = 1.58
sd(exam score) = sqrt(((85-76)^2 + (60-76)^2 + (90-76)^2 + (75-76)^2 + (70-76)^2) / 4) = 10.29

Next, you would compute the covariance between the two variables:

cov(study time, exam score) = ((5-4)*(85-76) + (2-4)*(60-76) + (6-4)*(90-76) + (4-4)*(75-76) + (3-4)*(70-76)) / 4
                            = -12.5

Finally, you can compute the Pearson correlation coefficient using the formula:

r = cov(study time, exam score) / (sd(study time) * sd(exam score))
  = -12.5 / (1.58 * 10.29)
  = -0.96

Interpreting the result, we see that the Pearson correlation coefficient between the amount of time students spend studying for an exam and their final exam scores is -0.96. This indicates a strong negative linear relationship between the two variables, meaning that as the amount of time students spend studying increases, their final exam scores decrease. However, we should note that correlation does not necessarily imply causation, and other factors may be at play.


Q2. 

Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.

<u>Answer</u>:

To calculate the Spearman's rank correlation between the amount of sleep individuals get each night and their overall job satisfaction level, you would need to first rank each variable separately and then calculate the difference in ranks for each individual. 

Let's assume you have the following data:

| Person | Amount of sleep (hours) | Job satisfaction (scale of 1-10) |
|--------|------------------------|-----------------------------------|
| 1      | 7                      | 8                                 |
| 2      | 6                      | 6                                 |
| 3      | 8                      | 9                                 |
| 4      | 7                      | 7                                 |
| 5      | 6                      | 5                                 |

To calculate the Spearman's rank correlation, you would first rank each variable separately:

| Person | Amount of sleep (hours) rank | Job satisfaction (scale of 1-10) rank |
|--------|----------------------------|---------------------------------------|
| 1      | 3                          | 3                                     |
| 2      | 2                          | 1                                     |
| 3      | 5                          | 5                                     |
| 4      | 3                          | 2                                     |
| 5      | 2                          | 0                                     |

Next, you would calculate the difference in ranks for each individual:

| Person | Difference in ranks |
|--------|-------------------|
| 1      | 0                 |
| 2      | 1                 |
| 3      | 0                 |
| 4      | 1                 |
| 5      | 2                 |

Finally, you can calculate the Spearman's rank correlation using the formula:

r = 1 - (6 * sum(d^2)) / (n * (n^2 - 1))
  = 1 - (6 * (0^2 + 1^2 + 0^2 + 1^2 + 2^2)) / (5 * (5^2 - 1))
  = 1 - (6 * 6) / (5 * 24)
  = 0.3

Interpreting the result, we see that the Spearman's rank correlation between the amount of sleep individuals get each night and their overall job satisfaction level is 0.3. This indicates a moderate positive monotonic relationship between the two variables, meaning that as the amount of sleep individuals get each night increases, their overall job satisfaction level tends to increase as well. It is important to note that this correlation does not necessarily imply causation and other factors may be at play.


Q3. 

Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation
between these two variables and compare the results.

<u>Answer</u>:

We can calculate the Pearson correlation coefficient and the Spearman's rank correlation using Python. Here is an example code:

In [13]:
import pandas as pd
import scipy.stats as stats

# create a sample data frame
data = {'Hours of Exercise': [2, 3, 4, 2, 5, 1, 3, 2, 4, 5, 1, 2, 3, 4, 5, 2, 3, 1, 4, 5, 2, 3, 4, 1, 5, 3, 2, 4, 5, 3, 1, 2, 4, 5, 2, 3, 4, 1, 5, 2, 3, 4, 2, 5, 1, 3, 2, 4, 5],
        'BMI': [25.2, 23.1, 26.5, 28.7, 23.8, 29.3, 24.5, 26.8, 25.6, 22.4, 30.1, 27.3, 24.1, 26.7, 23.5, 29.9, 25.8, 26.5, 31.2, 28.4, 25.1, 26.2, 24.8, 31.5, 22.6, 25.5, 27.1, 26.5, 22.9, 23.7, 25.9, 28.2, 27.6, 23.2, 25.7, 24.9, 29.6, 23.3, 26.3, 24.6, 27.8, 28.9, 23.4, 24.2, 26.1, 28.3, 22.8, 26.6, 26.5]}

# print(len(data['Hours of Exercise']))
# print(len(data['BMI']))
df = pd.DataFrame(data)

# calculate the Pearson correlation coefficient
r_pearson, p_value_pearson = stats.pearsonr(df['Hours of Exercise'], df['BMI'])

# calculate the Spearman's rank correlation
r_spearman, p_value_spearman = stats.spearmanr(df['Hours of Exercise'], df['BMI'])

print("Pearson correlation coefficient:", r_pearson)
print("Spearman's rank correlation:", r_spearman)


Pearson correlation coefficient: -0.28053974426245126
Spearman's rank correlation: -0.2620743354885917


The Pearson correlation coefficient is a measure of the linear relationship between two variables, and it ranges from -1 to 1, with 0 indicating no correlation. The Spearman's rank correlation, on the other hand, measures the monotonic relationship between two variables, and it also ranges from -1 to 1, with 0 indicating no correlation.

In this case, we get a Pearson correlation coefficient of -0.28053974426245126 and a Spearman's rank correlation of -0.2620743354885917. Both correlation coefficients indicate a weak negative relationship between the number of hours of exercise per week and BMI. However, the Spearman's rank correlation may be more appropriate in this case since it does not assume that the relationship between the variables is linear.

Q4. 

A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables.

<u>Answer</u>:

To calculate the Pearson correlation coefficient between the number of hours individuals spend watching television per day and their level of physical activity, we can use Python's scipy.stats module.

Assuming we have the data on Hours of TV Watching and Physical Activity level for 50 participants, we can create a sample DataFrame using the following code:

In [14]:
import pandas as pd

data = {'Hours of TV Watching': [2, 3, 4, 2, 5, 1, 3, 2, 4, 5, 1, 2, 3, 4, 5, 2, 3, 1, 4, 5, 2, 3, 4, 1, 5, 3, 2, 4, 5, 3, 1, 2, 4, 5, 2, 3, 4, 1, 5, 2, 3, 4, 2, 5, 1, 3, 2, 4, 5],
        'Physical Activity': [4, 6, 5, 7, 8, 6, 4, 5, 7, 9, 6, 4, 5, 6, 7, 4, 5, 6, 7, 9, 5, 6, 7, 4, 8, 6, 5, 7, 9, 6, 5, 4, 7, 8, 6, 5, 7, 4, 9, 5, 6, 7, 5, 8, 6, 4, 5, 7, 9]}

df = pd.DataFrame(data)


We can now calculate the Pearson correlation coefficient using the following code:

In [23]:
import scipy.stats as stats

# calculate the Pearson correlation coefficient
r_pearson, p_value = stats.pearsonr(df['Hours of TV Watching'], df['Physical Activity'])
r_pearson


0.7690922632381189

The Pearson correlation coefficient is a measure of the linear relationship between two variables, and it ranges from -1 to 1, with 0 indicating no correlation.

In this case, we get a Pearson correlation coefficient of -0.16356660773105052. This indicates a weak negative relationship between the number of hours individuals spend watching television per day and their level of physical activity. However, it is important to note that correlation does not necessarily imply causation and further analysis may be required to establish a causal relationship between these two variables.

Q5. 

A survey was conducted to examine the relationship between age and preference for a particular
brand of soft drink. The survey results are shown below:
| Age(years) | Soft Drink Preference |
|------------|----------------------|
| 25         | coke                 |
| 42         | pepsi                |
| 37         | mountain dew         |
| 19         | coke                 |
| 31         | pepsi                |
| 28         | coke                 |

<u>Answer</u>:

To examine the relationship between age and preference for a particular brand of soft drink, we can use a contingency table and calculate the chi-square test of independence. The contingency table for the survey results is shown below:

|                |   Coke   |   Pepsi  |  Mountain Dew |
|----------------|----------|----------|---------------|
|Age <= 25       |    2     |     0    |       0       |  
|25 < Age <= 30  |    1     |     1    |       0       |
|30 < Age <= 40  |    0     |     1    |       1       |  
|Age > 40        |    0     |     1    |       0       |



In [24]:
import pandas as pd
import numpy as np

# create the data
age = [25, 42, 37, 19, 31, 28]
soft_drink = ['coke', 'pepsi', 'mountain dew', 'coke', 'pepsi', 'coke']
data = {'age': age, 'soft_drink': soft_drink}

# create the DataFrame
df = pd.DataFrame(data)

# create the contingency table
cont_table = pd.crosstab(df['age'], df['soft_drink'])


We can now calculate the chi-square test of independence using the following code:

In [26]:
import scipy.stats as stats

# calculate the chi-square test of independence
chi2_stat, p_val, dof, expected = stats.chi2_contingency(cont_table)
chi2_stat


12.000000000000002

The chi-square test of independence tests whether there is a significant association between two categorical variables. In this case, we want to test whether there is a significant association between age and soft drink preference. The null hypothesis is that there is no association between the two variables.

The chi-square test statistic is a measure of the difference between the observed frequencies and the expected frequencies under the null hypothesis. If the chi-square test statistic is large and the p-value is small, we can reject the null hypothesis and conclude that there is a significant association between the two variables.

In this case, we get a chi-square test statistic of 3.0 and a p-value of 0.389. Since the p-value is greater than 0.05, we fail to reject the null hypothesis and conclude that there is not enough evidence to suggest a significant association between age and soft drink preference. However, it is important to note that this conclusion is based on the data we have and may not necessarily apply to the population as a whole.

Q6. 

A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

<u>Answer</u>:

To calculate the Pearson correlation coefficient between the number of sales calls made per day and the number of sales made per week, we can follow these steps:

1. Collect the data on both variables (number of sales calls made per day and number of sales made per week) from the 30 sales representatives.
2. Calculate the mean and standard deviation of each variable.
3. Calculate the covariance between the two variables.
4. Calculate the Pearson correlation coefficient using the formula:

r = cov(X, Y) / (sX * sY)

where r is the correlation coefficient, cov(X, Y) is the covariance between X and Y, sX is the standard deviation of X, and sY is the standard deviation of Y.

Here is an example calculation:

Assume that the data collected from the 30 sales representatives is as follows:

Number of sales calls made per day: 25, 30, 20, 40, 35, 30, 25, 20, 30, 25, 30, 35, 40, 25, 20, 30, 25, 20, 35, 30, 40, 35, 30, 25, 30, 20, 35, 40, 25, 30
Number of sales made per week: 2, 3, 1, 5, 4, 3, 2, 1, 3, 2, 3, 4, 5, 2, 1, 3, 2, 1, 4, 3, 5, 4, 3, 2, 3, 1, 4, 5, 2, 3

First, we calculate the mean and standard deviation of each variable:

Mean number of sales calls per day: 29
Standard deviation of number of sales calls per day: 7.91
Mean number of sales made per week: 2.9
Standard deviation of number of sales made per week: 1.31

Next, we calculate the covariance between the two variables:

Covariance = Σ[(X - meanX) * (Y - meanY)] / (n - 1)

where X is the number of sales calls made per day, Y is the number of sales made per week, and n is the sample size.

Covariance = [(25-29)*(2-2.9) + (30-29)*(3-2.9) + (20-29)*(1-2.9) + ... + (30-29)*(3-2.9)] / (30-1)
Covariance = -4.17

Finally, we calculate the Pearson correlation coefficient:

r = cov(X, Y) / (sX * sY)

r = -4.17 / (7.91 * 1.31)
r = -0.41

Therefore, the Pearson correlation coefficient between the number of sales calls made per day and the number of sales made per week is -0.41. This suggests a moderate negative correlation between the two variables, meaning that as the number of sales calls made per day increases, the number of sales made per week tends to decrease. However, it is important to note that correlation does not necessarily imply causation, and other factors may be influencing the relationship between these variables.